From d0114ff78c3a5611df24b0868dd9c49db7ce12ee Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Wed, 28 Aug 2024 14:00:57 +0200
Subject: [PATCH] arm64: dts: st: add ipcc2 node for Cortex-M0

Define ipcc2 node used by Cortex-M0.

Change-Id: I84ffbeafa30826a36a9bc60a1ede1c8d654ca232
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/400712
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 1eafc59ac426..cc49deda130c 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2982,6 +2982,18 @@ exti2: interrupt-controller@46230000 {
 				<&intc GIC_SPI 213 IRQ_TYPE_LEVEL_HIGH>;	/* EXTI_70 */
 		};
 
+		ipcc2: mailbox@46250000 {
+			compatible = "st,stm32mp1-ipcc";
+			#mbox-cells = <1>;
+			reg = <0x46250000 0x400>;
+			st,proc-id = <1>;
+			interrupts = <GIC_SPI 175 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 176 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "rx", "tx";
+			clocks = <&scmi_clk CK_SCMI_IPCC2>;
+			status = "disabled";
+		};
+
 		fmc: memory-controller@48200000 {
 			compatible = "st,stm32mp25-fmc2-ebi";
 			reg = <0x48200000 0x400>;
-- 
2.39.5

